#VRML_SIM R2021a utf8
WorldInfo {
  basicTimeStep 16
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -1 0 0 1.5707963267948966
  position -8.871898473247147e-17 2.557150634123821 1.4195037557195436e-16
  follow "robot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
CircleArena {
}
WoodenBox {
  translation 0.46 0.15 0
  rotation 0 1 0 0.523599
  size 0.3 0.3 0.3
}
WoodenBox {
  translation 0.0373486 0.15 0.51238
  rotation 0 1 0 -0.523595307179586
  name "wooden box(1)"
  size 0.3 0.3 0.3
}
WoodenBox {
  translation -0.098367 0.15 -0.66424
  rotation 0 -1 0 -0.2618053071795865
  name "wooden box(2)"
  size 0.3 0.3 0.3
}
Robot {
  translation 0 0.1 0
  children [
    DEF BODY Shape {
      appearance PBRAppearance {
        baseColor 1 0 1
        metalness 0
      }
      geometry Box {
        size 0.2 0.04 0.1
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor 0.05 0 0.06
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint Solid {
        translation 0.05 0 0.06
        rotation -0.9999999999999999 0 0 1.5707953071795862
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 1 0.666667 0
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor 0.05 0 -0.06
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint Solid {
        translation 0.05 0 -0.06
        rotation 1 0 0 1.5708
        children [
          USE WHEEL
        ]
        name "solid(1)"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor -0.05 0 0.06
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint Solid {
        translation -0.05 0 0.06
        rotation 1 0 0 1.5708
        children [
          USE WHEEL
        ]
        name "solid(2)"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 0 -1
        anchor -0.05 0 -0.06
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint Solid {
        translation -0.05 0 -0.06
        rotation 1 0 0 1.5708
        children [
          USE WHEEL
        ]
        name "solid(3)"
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    DistanceSensor {
      translation 0.1 0 -0.04
      rotation 0 1 0 0.261799
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "left distance sensor"
    }
    DistanceSensor {
      translation 0.1 0 0.04
      rotation 0 1 0 -0.2617996938995747
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 1 0
            metalness 0
          }
          geometry Box {
            size 0.01 0.01 0.01
          }
        }
      ]
      name "right distance sensor"
    }
  ]
  boundingObject USE BODY
  physics Physics {
  }
  controller "WheelRobot"
}
